sql/postgrest.sql

```diff
@@ -468,6 +468,8 @@ returns table (
   statement_timeout text
 ) language sql stable as
 $$
+ -- TODO: This query is very slow (almost a second compared to 30ms from other queries)
+ --       Needs to be optimized
```

Comment:
Is this the same on postgrest core?

Reply:
No, the query is fast on core. Some JOINS that I added could be slowing it.

Reply:
Yeah, these joins seem to be the culprit. An expensive Sequential Scan is returned when executing them.

```sql
      JOIN (pg_type t JOIN pg_namespace nt ON t.typnamespace = nt.oid)
        ON pa.type = t.oid
      LEFT JOIN (pg_type bt JOIN pg_namespace nbt ON bt.typnamespace = nbt.oid)
        ON t.typtype = 'd' AND t.typbasetype = bt.oid
      LEFT JOIN (pg_type t_arr JOIN pg_namespace nt_arr ON t_arr.typnamespace = nt_arr.oid)
        ON t.oid = t_arr.typarray
      LEFT JOIN (pg_type bt_arr JOIN pg_namespace nbt_arr ON bt_arr.typnamespace = nbt_arr.oid)
        ON t_arr.typtype = 'd' AND t_arr.typbasetype = bt_arr.oid
```

Reply:
Using a CROSS JOIN helped in avoiding the loops, it takes ~160ms instead of ~1000ms now. Better, but still should be faster.

```sql
    FROM pg_proc p
      CROSS JOIN unnest(proargnames, proargtypes, proargmodes)
        WITH ORDINALITY AS pa (name, type, mode, idx)
```